x/datadeal/keeper/deal.go

```diff
@@ -149,8 +149,14 @@ func (k Keeper) SellData(ctx sdk.Context, msg *types.MsgSellData) error {
 		return err
 	}
 
-	//TODO: Check the Deal which seller want to sell data to the deal exists
-	//k.GetDeal()
+	deal, err := k.GetDeal(ctx, msg.DealId)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrSellData, err.Error())
+	}
+
+	if deal.Status == types.DEAL_STATUS_INACTIVE {
+		return sdkerrors.Wrapf(types.ErrSellData, "deal status is INACTIVE")
+	}
```

Comment:
You may have to check if the deal is sold out (or completed?) which means `CurNumData` reaches to `MaxNumData`.
For that, we need a new status for deal accordingly.

Reply:
I think that when the `curNumData` reaches to `maxNumData`, the deal's status will be change to `INACTIVE`.
As you say, I agree that add a deal's completed status. So, I add a `COMPLETED` status and re-request your review.
Also, It need to implement well when update a deal's status in endblocker as I written in #414 .

Reply:
I missed the issue you mentioned. BTW, let's be clear because a completed deal is different from a deactivated deal.

Reply:
Applied in [1556530](https://github.com/medibloc/panacea-core/pull/432/commits/1556530df0025eb425665521576fe51f45776e70).